### ISSUE TITLE
vue+ts, when props are passed, the same type check error is reported

### ISSUE THREAD

**Issue by krytes** [microsoft/TypeScript#60676]
### Acknowledgement

- [x] I acknowledge that issues using this template may be closed without further explanation at the maintainer's discretion.

### Comment

1. Conditions
`ts + vue3 + naive-ui + pnpm + workspace`
`pnpm: 8.6.7  node: v16.17.0`

2. Display
![Image](https://github.com/user-attachments/assets/1187b6f8-efd8-457d-9897-5bda3b1066d6)

3. Code:

`test.vue ↓↓↓`
```
<template>
  <div>
    <NDataTable remote :columns="dialogColumns" />
    <NDataTable remote :columns="pageData.dialogColumns" />
  </div>
</template>

<script setup lang="tsx">
import { NDataTable } from 'naive-ui'
import { TableColumns } from 'naive-ui/es/data-table/src/interface'

const pageData = reactive({
  dialogColumns: [] as TableColumns<any>,
})

const dialogColumns = computed((): TableColumns<any> => pageData.dialogColumns)
</script>

<style scoped lang="less"></style>
```

`apps/web-doctor-adm
-package.json  ↓↓↓`
```
"eslint": "^8.56.0",
"@typescript-eslint/parser": "^6.18.1",
"eslint": "^8.56.0",
"eslint-config-prettier": "^9.1.0",
"eslint-plugin-import": "^2.29.1",
"eslint-plugin-prettier": "^5.1.3",
"eslint-plugin-vue": "^9.21.1",
"prettier-eslint": "^16.3.0",
"vue-eslint-parser": "^9.4.2",
"@types/js-cookie": "^3.0.2",
"@types/lodash": "^4.14.191",
"@types/qs": "^6.9.7",
"@typescript-eslint/eslint-plugin": "^6.18.1",
"@typescript-eslint/parser": "^6.18.1",
"typescript": "5.2.2",
```
`apps/web-doctor-adm
-tsconfig.json  ↓↓↓`
```
{
  "compilerOptions": {
    "allowJs": true,
    "target": "ES2020",
    "useDefineForClassFields": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "jsx": "preserve",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "esModuleInterop": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "skipLibCheck": true,
    "noEmit": true,
    "baseUrl": ".",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": false,

    "strict": true,
    "noUnusedLocals": false,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true,

    "allowImportingTsExtensions": true,

    "paths": {
      "@/*": ["src/*"]
    },
    "types": ["vite/client"]
  },
  "include": [
    "./components.d.ts",
    "src/**/*.ts",
    "src/**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "mock/**/*.ts"
  ],
  "references": [{ "path": "./tsconfig.node.json" }]
}

```

`root 
-package.json  ↓↓↓
`
```
"@typescript-eslint/eslint-plugin": "^6.18.1",
"@typescript-eslint/parser": "^6.18.1",
"eslint": "^8.56.0",
"eslint-config-prettier": "^9.1.0",
"eslint-plugin-import": "^2.29.1",
"eslint-plugin-prettier": "^5.1.3",
"eslint-plugin-vue": "^9.21.1",
"prettier": "^3.2.5",
"prettier-eslint": "^16.3.0",
"typescript": "^5.2.2",
```

4. Confuse
I don't know why it behaves like this. 
This is migrated from an independent project, and the original project does not report an error. pnpm-lock.yaml cannot be imported into the workspace. In fact, I tried to spend a lot of time modifying the pnpm-loc.yaml file to fix the problem, but after adjusting the version, it still reports an error.

5. Thanks for your help

**Comment by nmain** [comment]
Can you condense this into a minimal self-contained reproduction?  The fact that it did not occur in your original project suggests some package shenanigans, possibly pnpm-related, outside of Typescript's control.

**Comment by RyanCavanaugh** [comment]
We don't control the vue extension. Please provide a new issue that doesn't require vue if you'd like us to investigate. Thanks!
